lib/src/misc/frame_fit.dart

```diff
@@ -0,0 +1,204 @@
+import 'dart:math' as math;
+
+import 'package:flutter/widgets.dart';
+import 'package:flutter_map/src/geo/latlng_bounds.dart';
+import 'package:flutter_map/src/map/flutter_map_frame.dart';
+import 'package:flutter_map/src/misc/point.dart';
+import 'package:flutter_map/src/misc/private/bounds.dart';
+import 'package:latlong2/latlong.dart';
+
+abstract class FrameFit {
```

Comment:
Can't this be converted to a `sealed` class ?

Reply:
Yeah you're probably right. I mean a user could implement their own FrameFit if they wanted to but is that really likely?

Reply:
@TesteurManiak On further consideration I don't think this should be a sealed class. The interface of the class is safe in the sense that it can only calculate a non-nullable MapCamera result. If the user wishes to implement their own CameraFit it shouldn't cause any issues.

I will resolve this for now but if you disagree let me know!

Reply:
No problem, I'm okay with it staying an `abstract class` 👍